src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,533 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
         )
 
 
+class PlantsSmallFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """A table transformer specific to the ``plants_small_ferc1`` table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.PLANTS_SMALL_FERC1
+
+    @cache_df(key="main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Table specific transforms for plants_small_ferc1.
+
+        Params:
+            df: Pre-processed, concatenated XBRL and DBF data.
+
+        Returns:
+            A single transformed table concatenating multiple years of cleaned data
+            derived from the raw DBF and/or XBRL inputs.
+        """
+        df = (
+            self.normalize_strings(df)
+            .pipe(self.nullify_outliers)
+            .pipe(self.convert_units)
+            .pipe(self.categorize_strings)
+            .pipe(self.extract_ferc1_license)
+            .pipe(self.label_row_type)
+            .pipe(self.map_header_fuel_types)
+            .pipe(self.map_plant_name_fuel_types)
+            .pipe(self.associate_notes_with_values)
+            # .pipe(self.drop_invalid_rows)
```

Comment:
Figure I might add this in later, but I wasn't 100% on what it would be yet.

Reply:
These header/note methods don't drop those rows do they? Probably whether or not those rows are dropped we should be applying `drop_invalid_rows`. Presumably, the invalid-row dropper would probably drop all the header/notes rows because we will probably want to use a similar pattern of finding the rows that have nulls in all of the data columns. `drop_invalid_rows` is pretty flexible in that you can either give it columns that must be valid (i.e. the "data" columns) or the rows to ignore while checking for invalid data (i.e. the "id-ish" columns)